app/chat/[id].tsx

```
@@ -91,6 +93,33 @@ export default function Chat() {
+                    tracking.event('chat_create_user_goal');
+                    return formatGoalForAI(goal);
+                }
+                case 'update-user-goal': {
+                    const args = toolCall.args as typeof tools['update-user-goal']['parameters']['_type'];
+                    const updatedGoals = await updateGoalAndSave(args.id, args);
+                    if (!updatedGoals) {
+                        return `Goal #${args.id} not found. Max ID: ${goals.length}.`;
+                    } else {
+                        setGoals(updatedGoals);
+                        tracking.event('chat_update_user_goal');
+                        return formatGoalForAI(updatedGoals.find(g => g.id === args.id)!);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)